kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala

```diff
@@ -140,15 +142,17 @@ object TrinoContext {
   def buildTrinoResponse(qr: QueryResults, trinoContext: TrinoContext): Response = {
     val responseBuilder = Response.ok(qr)
 
-    trinoContext.catalog.foreach(
-      responseBuilder.header(TRINO_HEADERS.responseSetCatalog, _))
-    trinoContext.schema.foreach(
-      responseBuilder.header(TRINO_HEADERS.responseSetSchema, _))
+    // Note, We have injected kyuubi session id to session context so that the next query can find
+    // the previous session to restore the query context.
+    // It's hard to follow the Trino style that set all context to http headers.
+    // Because we do not know the context at server side. e.g. `set k=v`, `use database`.
+    // We also can not inject other session context into header before we supporting to map
+    // query result to session context.
```

Comment:
cc @iodone  
We can not inject session context using the current request header. e.g. the current query is `set k=v2` but he exists request header is `k=v`. So we only inject kyuubi session id to help next query find session context.

Reply:
Instead of returning all session information to the client with HTTP response header, set it to sessionConfig and use kyuubi session to save the session information in server side?

Reply:
yea, I think it's not the trino way but more suitable for Kyuubi

Reply:
Nice, elegant solutions